meson.build

```diff
@@ -4,12 +4,22 @@ project('Hyprland', 'cpp', 'c',
 
 add_global_arguments('-std=c++23', language: 'cpp')
 
+GIT_BRANCH = run_command('gitinfo.sh', 'GIT_BRANCH').stdout().strip()
+GIT_COMMIT_HASH = run_command('gitinfo.sh', 'GIT_COMMIT_HASH').stdout().strip()
+GIT_COMMIT_MESSAGE = run_command('gitinfo.sh', 'GIT_COMMIT_MESSAGE').stdout().strip()
+GIT_DIRTY = run_command('gitinfo.sh', 'GIT_DIRTY').stdout().strip()
+
 add_project_arguments(
   [
     '-Wno-unused-parameter',
     '-Wno-unused-value',
     '-Wno-missing-field-initializers',
     '-Wno-narrowing',
+
+    f'-DGIT_BRANCH="@GIT_BRANCH@"',
+    f'-DGIT_COMMIT_HASH="@GIT_COMMIT_HASH@"',
+    f'-DGIT_COMMIT_MESSAGE="@GIT_COMMIT_MESSAGE@"',
+    f'-DGIT_DIRTY="@GIT_DIRTY@"',
```

Comment:
If you're going to use Meson's documented support for f-strings, which was added in Meson 0.58, you might want to set that in the `project()` function as `meson_version: '>=0.58.0'` so that Meson can error out on older versions and tell the user to "upgrade Meson".

It's a pretty recent version and e.g. not available in Debian stable's official repositories. It's of course trivial to upgrade it with `pip install`, but it's quite easy to imagine people trying the version from the Debian repositories first, then having it fail, so letting them know the reason is because they have an old version of Meson is useful information.